src/binder/expression_visitor.cpp

```diff
@@ -84,13 +84,37 @@ expression_vector ExpressionChildrenCollector::collectRelChildren(const Expressi
     return result;
 }
 
-bool ExpressionVisitor::hasExpression(
+bool ExpressionVisitor::needFold(const Expression& expression) {
+    if (expression.expressionType == ExpressionType::LITERAL) {
```

Comment:
inline this return?
return expression.expressionType == ExpressionType::LITERAL ? return false : return isConstant(expression);